x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/popover_editor.tsx

```diff
@@ -56,6 +56,26 @@ function asOperationOptions(operationTypes: OperationType[], compatibleWithCurre
     }));
 }
 
+const LabelInput = ({ value, onChange }: { value: string; onChange: (value: string) => void }) => {
+  const [inputValue, setInputValue] = useState(value);
+  const onChangeDebounced = _.debounce(onChange, 256);
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const val = String(e.target.value);
+    setInputValue(val);
+    onChangeDebounced(val);
+  };
+
+  return (
+    <EuiFieldText
+      compressed
+      data-test-subj="indexPattern-label-edit"
+      value={inputValue}
```

Comment:
It seems like this is only updating the value of the actual input field if the user types something in it - props changes coming in after the component got rendered the first time will be ignored.

A `useEffect` updating the local state when the global state management pushes an update down should work fine. This is a very similar case handling it the same way: https://github.com/elastic/kibana/pull/69672/files

Reply:
Done! Thanks!